### PR TITLE
Pepa ps 1553 docs

### PR DESCRIPTION
**Pull request by joker806**
https://keboola.atlassian.net/browse/PS-1553
Based on https://github.com/keboola/staging-provider/pull/6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,97 @@
...
+// now the $strategyFactory is ready to be used
+$reader = new Reader($strategyFactory);
+```
+
+We start by creating a `StrategyFactory` needed by the reader. The strategy itself has no knowledge of which storage
+should be used with each staging type. This is what provider initializer does - configure the `StrategyFactory` for
+a specific type of staging.
+
+To create a provider initializer we pass it:
+* the `StragetyFactory` to initialize
```
Comment (odinuv) [inline README.md:54]:
```suggestion
* the `StrategyFactory` to initialize
```

File: README.md [hunk 2]
```diff
@@ -2,6 +2,97 @@
...
+The example above presents usage of `InputProviderInitializer` for configuration of input mapping `StrategyFactory` for
+a `Reader`. Similarly, we can use `OutputProviderInitializer` to configure output mapping `StrategyFactory` for a `Writer`. 
+
+## Internals
+The main objective of the library is to configure `StrategyFactory` so it knows which staging provider to
+use with each kind of storage.
+
+### Staging
+Generally there are two kind of staging:
+* local staging - used to store data locally on filesystem, represented `LocalStaging` class
```
Comment (odinuv) [inline README.md:75]:
```suggestion
* local staging - used to store data locally on filesystem, represented by `LocalStaging` class
```

**Review by odinuv — APPROVED** [review]
dobry!

**Review by pivnicek — APPROVED** [review]
this is great!

**Pull request merged**